File: van_der_pol/van_der_pol.jl
using OrdinaryDiffEq

function system(du, u, p, t)
    x, y = u
    μ = p[1]
    du[1] = y
    du[2] = μ * (1 - x^2) * y - x
end

function generate_dataset(num_simulations, t_span, params)
    tspan = (t_span[1], t_span[2])
    t = range(tspan[1], tspan[2], length=200)
    
    initial_conditions = rand(num_simulations, 2) .* 10 .- 5
    dataset = []
    
    for ic in eachrow(initial_conditions)
        prob = ODEProblem(system, ic, tspan, params)
        sol = solve(prob, Tsit5(), saveat=t)
        push!(dataset, Array(sol))
    end
    
    return cat(dataset..., dims=3), collect(t)
end

File: van_der_pol/van_der_pol.py
import numpy as np
from scipy.integrate import odeint

def system(state, t, mu):
    x, y = state
    dxdt = y
    dydt = mu * (1 - x**2) * y - x
    return [dxdt, dydt]

def simulate(initial_conditions, t, params):
    mu = params[0]
    solutions = []
    for ic in initial_conditions:
        solution = odeint(system, ic, t, args=(mu,))
        solutions.append(solution)
    return np.array(solutions)

def generate_dataset(num_simulations, t_span, params):
    np.random.seed(42)
    t = np.linspace(t_span[0], t_span[1], 200)
    initial_conditions = np.random.rand(num_simulations, 2) * 4 - 2
    dataset = simulate(initial_conditions, t, params)
    return dataset, t